pipeline/environment_function_test.go

```diff
@@ -0,0 +1,132 @@
+package pipeline
+
+import (
+	"errors"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/goreflect/gostructor/infra"
+	"github.com/goreflect/gostructor/tags"
+)
+
+func TestEnvironmentConfig_GetBaseType(t *testing.T) {
+	myStruct := struct {
+		field1 string `cf_env:"testBaseType"`
+	}{}
+	os.Setenv("testBaseType", "tururu")
+	myStruct1 := reflect.Indirect(reflect.ValueOf(myStruct))
+	field1 := myStruct1.Type().Field(0)
+	t.Log("type of field: ", field1.Name+", "+field1.Type.Name())
+	type args struct {
+		context *structContext
+	}
+	tests := []struct {
+		name   string
+		config EnvironmentConfig
+		args   args
+		want   infra.GoStructorValue
+	}{
+		{
+			name:   "get success base type string from environment",
+			config: EnvironmentConfig{},
+			args: args{
+				context: &structContext{
+					StructField: field1,
+					Value:       myStruct1.Field(0),
+				},
+			},
+			want: infra.NewGoStructorTrueValue(reflect.ValueOf("tururu")),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			config := EnvironmentConfig{}
+			got := config.GetBaseType(tt.args.context)
+			os.Remove("testBaseType")
```

Comment:
Use defer

Reply:
why? Using only for using it's no valid way for me.

Reply:
in this case it is unjustified use, or prove please your view point of using this keyword in this place

Reply:
defer is the only right way to finalize using of resource that must be somehow finalized. No matter how much time will pass after writing the initial code or how many changes you will do, you will never lose the connection between opening any resource and its closing or setting and removing if they are placed right one after another.
What's more, you are removing it in the for loop, it doesn't make any sense. You won't be able to use it in the following iteration.

Reply:
I went to check what's going to happen when you remove an nonexisting env and it turned out that we are using the wrong function. [Remove](https://golang.org/pkg/os/#Remove) removes a file, what we need is [Unsetenv](https://golang.org/pkg/os/#Unsetenv) and it returns error when you removes nonexisting env.

Reply:
++